inContainer false positive on DigitalOcean Droplet

### What is the issue?

`inContainer` should not report true on DigitalOcean Ubuntu 18.04 Droplet.

### Steps to reproduce

Adding this diff to `hostinfo/hostinfo_linux_test.go`:

```
diff --git a/hostinfo/hostinfo_linux_test.go b/hostinfo/hostinfo_linux_test.go
index 4859167a..e94c91e2 100644
--- a/hostinfo/hostinfo_linux_test.go
+++ b/hostinfo/hostinfo_linux_test.go
@@ -34,3 +34,9 @@ func TestQnap(t *testing.T) {
                t.Errorf("got %q; want %q", got, want)
        }
 }
+
+func TestIncontainer(t *testing.T) {
+       if lazyInContainer.Get().EqualBool(true) {
+               t.Error("not in container, but report true")
+       }
+}
```

Then compiling the test, copy to a DigitalOcean Ubuntu 18.04 Droplet:

```
$ cd hostinfo
$  CGO_ENABLED=0 go test -c .
```

Running the test in Droplet:

```
$ ./hostinfo.test -test.run=TestIncontainer -test.v
=== RUN   TestIncontainer
    hostinfo_linux_test.go:40: not in container, but report true
--- FAIL: TestIncontainer (0.00s)
FAIL
```

### Are there any recent changes that introduced the issue?

I think https://github.com/tailscale/tailscale/pull/1643 introduced this.

### OS

Linux

### OS version

_No response_

### Tailscale version

v1.44.0

### Other software

_No response_

### Bug report

_No response_

--- a/hostinfo/hostinfo.go
+++ b/hostinfo/hostinfo.go
@@ -283,7 +283,7 @@ func inContainer() opt.Bool {
 		return nil
 	})
 	lineread.File("/proc/mounts", func(line []byte) error {
-		if mem.Contains(mem.B(line), mem.S("fuse.lxcfs")) {
+		if mem.Contains(mem.B(line), mem.S("lxcfs /proc/cpuinfo fuse.lxcfs")) {
 			ret.Set(true)
 			return io.EOF
 		}

